app/payloads/api/v1/spec.rb

```diff
@@ -61,85 +61,117 @@ def phone_number_spec
-  def patient_error_spec
+  def self.patient_sync_from_user_errors_spec
     { type:       :object,
       properties: {
-        id:            { type: :string, format: :uuid },
-        address:       { '$ref' => '#/definitions/error_spec' },
-        phone_numbers: { type:  :array,
-                         items: { '$ref' => '#/definitions/error_spec' } } },
-      required:   %w[id] }
+        errors: {
+          type:  :array,
+          items: { '$ref' => '#/definitions/error_spec' } } } }
   end
 
-  def patient_sync_from_user_errors_spec
+  def self.error_spec
     { type:       :object,
       properties: {
-        errors: {
-          type:  :array,
-          items: { '$ref' => '#/definitions/patient_error_spec' } } } }
+        id:               { type:        :string,
+                            format:      :uuid,
+                            description: 'Id of the record with errors' },
+        schema:           { type:        :array,
+                            items:       { type: :string },
+                            description: 'List of json schema error strings describing validation errors' },
+        field_with_error: { type:  :array,
+                            items: { type: :string } } } }
   end
 
-  def patient_sync_to_user_request_spec
+  def self.patient_sync_to_user_request_spec
     [processed_since.merge(in: :query),
      { in:          :query, name: :limit, type: :integer,
        description: 'Number of record to retrieve (a.k.a batch-size)' }]
   end
 
-  def patient_sync_from_user_request_spec
+  def self.patient_sync_from_user_request_spec
     { type:       :object,
       properties: {
         patients: { '$ref' => '#/definitions/nested_patients' } },
       required:   %w[patients] }
   end
 
-  def patient_sync_to_user_response_spec
+  def self.patient_sync_to_user_response_spec
     { type:       :object,
       properties: {
         patients:        { '$ref' => '#/definitions/nested_patients' },
         processed_since: { '$ref' => '#/definitions/processed_since' } } }
   end
 
-  def all_definitions
+  def self.all_definitions
     { timestamp:          timestamp,
       nullable_timestamp: nullable_timestamp,
       processed_since:    processed_since,
       patient:            patient_spec,
       address:            address_spec,
       phone_number:       phone_number_spec,
       phone_numbers:      phone_numbers_spec,
-      error_spec:         error_spec,
-      patient_error_spec: patient_error_spec,
-      nested_patients:    nested_patients }
+      nested_patient:     nested_patient,
+      nested_patients:    nested_patients,
+      error_spec: error_spec }
+  end
+
+  def self.swagger_info
+    {
+      description: I18n.t('api_description'),
+      version:     'v1',
+      title:       'RedApp Server',
+      'x-logo'     => {
+        url:             'https://static1.squarespace.com/static/59945d559f7456b755d759f2/t/59aebc5ecf81e0ac9b4b6f59/1526304079797/?format=1500w',
```

Comment:
Is it possible to use an asset in the repo for this?

Reply:
Moved the strings to locale/en.yml for now. This would be convenient for internationalizing our API docs.

Reply:
Oops, I misunderstood this.
Added the logo as an image asset.